aspnetcore/blazor/components/css-isolation.md

```diff
@@ -0,0 +1,160 @@
+
+## Child component support
+
+By default, CSS isolation only applies to the component you associate with `MyComponent.razor.css`. To apply changes to a child component, use the `::deep` combinator to any descendant elements in the parent component's `razor.css` file. The `::deep` combinator selects elements that are *descendants* of an element's generated scope identifier. 
+
+The following example shows a parent component, called `ChildExample` in a file called `Pages/ChildExample.razor`, with a child component called `MyChild`:
+
+`ChildExample.razor`:
+```razor
+@page "/child-example"
+
+<div>
+    <h1>Child Example</h1>
+    <MyChild />
+</div>
+```
+
+`MyChild.razor`:
+```razor
+@page "/first-child"
+
+<h1>A Child Component</h1>
+```
+
+Update the `h1` declaration in `MyComponent.razor.css` with the `::deep` combinator to signify the `h1` style declaration must apply to the parent component and its children.
+
+```css
+::deep h1 { 
+    color: red;
+}
+```
+
+The `h1` style now applies to the `ChildExample` and `FirstChild` components without the need to create separate scoped CSS files for the child components.
+
+> [!NOTE]
+> The `::deep` combinator works only with descendant elements. The following HTML structure applies the `h2` styles to components as expected:
+> 
+>```razor
+><div>
+>    <h1>Child Example</h1>
+>    <MyChild />
+></div>
+>```
+> In this scenario, ASP.NET Core applies the parent component's scope identifier to the `div` element, so it knows to inherit styles from the parent component.
+>
+>However, excluding the `div` element removes the descendant relationship, and the style will *not* be applied to the child component. 
+>
+>```razor
+><h1>Child Example</h1>
+><MyChild />
+>```
+
+## CSS preprocessor support
+
+CSS preprocessors are useful with improving CSS development by utilizing features like variables, nesting, modules, mixins, and inheritance. While CSS isolation does not *natively* support CSS preprocessors such as Sass or Less, integrating them is seamless as long as preprocessor compilation occurs before Blazor rewrites the CSS selectors during the build process.
+
+For example, in the Visual Studio Task Runner Explorer, configure existing preprocessor compilation as a **Before Build** task.
+
+Many third-party NuGet packages, such as [Delegate.SassBuilder](https://www.nuget.org/packages/Delegate.SassBuilder/), can compile SASS/SCSS files at the beginning of the build process before CSS isolation occurs—preventing any additional configuration.
+
+For example, when you add a `MyComponent.razor.scss` file, these tools can compile SASS changes to `MyComponent.razor.css` at the beginning of the build process. With that step complete, the isolation step is ready. 
+
+## CSS isolation configuration
+
+When working with isolated CSS, no configuration is required. However, configuration is available for advanced scenarios or when there are dependencies on other existing tools or workflows.
```

Comment:
This first sentence feels a little weird in conjunction with the next. Maybe say something like "CSS isolation is designed to work out-of-the-box, but provides configuration for some advanced scenarios..."

Reply:
Good point. Changed to:

CSS isolation is designed to work out-of-the-box, but provides configuration for some advanced scenarios—such as when there are dependencies on existing tools or workflows.